nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/RecordField.java

```diff
@@ -76,8 +77,12 @@ public RecordField(final String fieldName, final DataType dataType, final Object
         } else {
             this.aliases = Collections.unmodifiableSet(aliases);
         }
-
-        this.defaultValue = defaultValue;
+        if (RecordFieldType.ARRAY.equals(dataType.getFieldType()) && (defaultValue instanceof String
+                || defaultValue instanceof Number || defaultValue instanceof Boolean)) {
```

Comment:
This will cause [TestAvroReaderWithEmbeddedSchema](https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/test/java/org/apache/nifi/avro/TestAvroReaderWithEmbeddedSchema.java) to fail.

Reply:
Yes, it solves the reported issue, but it makes the test fails. It is required to think another solution approach. It is not as simple I initially thought